config/initializers/session_store.rb

```diff
@@ -1,4 +1,6 @@
 # frozen_string_literal: true
 # Be sure to restart your server when you modify this file.
 
-Rails.application.config.session_store :cookie_store, key: "_velum_session"
+Rails.application.config.session_store :cookie_store,
+  key:    "_velum_session",
+  secure: Rails.env.production?
```

Comment:
> We should stop trying to make dev/test/prod environments different.

They are different by nature. Rails will reload classes on `development`, while it won't on `production`, logging is also different as well as lots of other small (and big) details. We can try to converge in this special case, but we just cannot make dev/test/prod environments to be the same, mainly because they are not meant to be the same.

> Why would setting ^ cause test environments to fail? Do they still access non https endpoints?

Tests are completely run with `http`, no `https` at all: https://github.com/kubic-project/velum/blob/927bc4972aa969ea505f3631b2a3cd2b5806ae46/app/controllers/application_controller.rb#L5. Also, calls to salt-api: https://github.com/kubic-project/velum/blob/927bc4972aa969ea505f3631b2a3cd2b5806ae46/lib/velum/salt_api.rb#L82-L86

I haven't looked in detail why tests fail, but if we enable `https` they would for sure be orders of magnitude slower, do we want them to be slower for controller testing over `https`? Don't we have other tests for that? I'm wondering here, open to discussion. I'd expect this tests to be as fast as possible, so you can run them several times while you are hacking on something, I don't want tests to be slow because they are using `https` in an environment not meant for safety but speed.

I really believe it's not a matter of having all environments in the same way, but to have the right tool with the right settings for the job.

My 2 cents.